grails-app/controllers/org/pih/warehouse/inventory/InventoryItemController.groovy

```diff
@@ -242,54 +237,63 @@ class InventoryItemController {
     }
 
 
-    def showPending = {
+    def showPendingInbound = {
 
         Product product = Product.get(params.id)
         Location location = Location.get(session?.warehouse?.id)
-        StockMovementType stockMovementType = params.type as StockMovementType
-        def itemsMap, requisitionItems, shipmentItems, orderItems = []
-
-        if (!stockMovementType) {
-            throw new IllegalArgumentException("Stock movement type is required")
+        def itemsMap = [:], requisitionItems, shipmentItems, orderItems = []
+
+        shipmentItems = shipmentService.getPendingInboundShipmentItems(location, product)
+        shipmentItems.sort { it.shipment.currentStatus }.groupBy { it.shipment }.collect { k, v ->
+            itemsMap.put(k, [
+                    quantityRemaining: v.quantityRemaining.sum(),
+                    quantityPurchased: 0,
+                    shipDate: k.expectedShippingDate,
+                    type: 'Stock Movement'
+            ]
+            )
+        }
+        orderItems = orderService.getPendingInboundOrderItems(location, product)
+        orderItems.collect {
+            def existingItem = itemsMap.find {k, v -> k instanceof OrderItem && k.actualReadyDate == it.actualReadyDate && k.order == it.order}
+            if (!existingItem) {
+                itemsMap.put(it, [
+                        quantityRemaining: 0,
+                        quantityPurchased: (it.quantityRemaining * it.quantityPerUom).toInteger(),
+                        shipDate         : it.actualReadyDate,
+                        type             : 'Purchase Order',
+                ]
+                )
+            } else {
+                itemsMap[existingItem.getKey()].quantityPurchased += (it.quantityRemaining * it.quantityPerUom).toInteger()
+            }
         }
 
-        Location origin = stockMovementType == StockMovementType.INBOUND ? null : location
-        Location destination = stockMovementType == StockMovementType.OUTBOUND ? null : location
 
-        if (origin) {
-            requisitionItems = requisitionService.getPendingRequisitionItems(origin, product)
-            itemsMap = requisitionItems.groupBy { it.requisition }
-        } else if (destination) {
-            shipmentItems = shipmentService.getPendingInboundShipmentItems(destination, product)
-            itemsMap = shipmentItems.sort { it.shipment.currentStatus }.groupBy { it.shipment }
-            orderItems = orderService.getPendingInboundOrderItems(destination, product)
-            itemsMap += orderItems.groupBy { it.order }
+        log.info "itemsMap: " + itemsMap
+
+        render(template: "showPendingInboundStock", model: [product: product, itemsMap: itemsMap])
+    }
+
+    def showPendingOutbound = {
+        Product product = Product.get(params.id)
+        Location location = Location.get(session?.warehouse?.id)
+        def itemsMap = [:], requisitionItems
```

Comment:
Split this into two lines for clarity.
`def itemsMap = [:]`
`def requisitionItems`